src/freertos_drivers/sam/FeatherM4Can.hxx

```diff
@@ -0,0 +1,249 @@
+/** \copyright
+ * Copyright (c) 2025, Balazs Racz
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ *  - Redistributions of source code must retain the above copyright notice,
+ *    this list of conditions and the following disclaimer.
+ *
+ *  - Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ * \file FeatherCan.hxx
+ *
+ * Wrapper for Arduino CAN driver for the Feather M4 CAN board using the
+ * external ACANFD_FeatherM4CAN library.
+ *
+ * @author Balazs Racz
+ * @date 29 July 2025
+ */
+
+#ifndef _ARDUINO_SAM_FEATHERM4CAN_HXX_
+#define _ARDUINO_SAM_FEATHERM4CAN_HXX_
+
+#ifndef ARDUINO_FEATHER_M4_CAN
+#error "This code only works on the Arduino Feather M4 CAN (SAME51)"
+#endif
+
+#include <ACANFD_FeatherM4CAN-from-cpp.h>
+
+#include "freertos_drivers/arduino/Can.hxx"
+
+/// Wrapper for Arduino CAN driver for the Feather M4 CAN board using the
+/// external ACANFD_FeatherM4CAN library.
+/// Usage:
+///
+/// // These are required to get the right CAN setup working.
+/// #define CAN0_MESSAGE_RAM_SIZE (0)
+/// #define CAN1_MESSAGE_RAM_SIZE (1728)
+/// #include <ACANFD_FeatherM4CAN.h>
+///
+/// #include <OpenMRNLite.h>
+/// FeatherM4Can can_driver;
+///
+/// void setup() {
+///   ...
+///   HASSERT(can_driver.begin());
+///   openmrn.add_can_port(&CanDriver);
```

Comment:
should this be `CanDriver` or `can_driver`?

Reply:
fixed